jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/DatasourceDataFetcher.java

```diff
@@ -0,0 +1,125 @@
+            List<String> assetGroups = getVisibleAssetGroupsFiltered(aliases);
+            datasourceData.setAssetGroups(assetGroups);
+            datasourceData.setAccountIds(getAccountsByDatasource(datasource));
+            return datasourceData;
+        } catch (Exception e) {
+            LOGGER.error("Error while constructing datasource related data", e);
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves a list of unique alias names associated with a list of Elasticsearch indices.
+     * Sends HTTP requests to Elasticsearch to fetch alias information and stores the unique alias names.
+     *
+     * @return A list of unique alias names associated with the specified indices.
+     */
+    private List<String> getAliasByDatasource(String datasource) {
+        try {
+            return fetchAliases().entrySet().stream()
+                    .filter(entry -> entry.getValue().stream().anyMatch(value -> value.startsWith(datasource + "_")))
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            LOGGER.error("An error occurred while retrieving alias names: " + e.getMessage(), e);
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Retrieves a list of visible asset group names that match a given list of asset group names.
+     *
+     * @param assetListToFilter A list of asset group names to filter by visibility.
+     * @return A list of visible asset group names that match the provided asset list.
+     */
+    private List<String> getVisibleAssetGroupsFiltered(List<String> assetListToFilter) {
+        String query = "select distinct groupName from cf_AssetGroupDetails where isVisible = true and groupName in " +
+                "('" + String.join("','", assetListToFilter) + "')";
+        return  RDSDBManager.executeStringQuery(query);
+    }
+
+    /**
+     * Retrieves a list of account IDs associated with a specific datasource.
+     *
+     * @param datasource The name of the datasource for which account IDs are retrieved.
+     * @return A list of account IDs that meet the criteria.
+     */
+    private List<String> getAccountsByDatasource(String datasource) {
+        String query = "select accountId from cf_Accounts  where platform = '" +
+                datasource + "' and accountStatus= 'configured'";
+        return RDSDBManager.executeStringQuery(query);
+    }
+
+    /**
+     * Fetches aliases from Elasticsearch and returns a mapping of aliases to their associated indices.
+     *
+     * @return A mapping of aliases to their associated indices.
+     * @throws IOException If an error occurs during the retrieval of aliases.
+     */
+    private Map<String, List<String>> fetchAliases() throws Exception {
+        String urlToQuery = ES_URI + "/_cat/aliases";
+        String responseDetails = HttpUtil.httpGetMethodWithHeaders(urlToQuery, new HashMap<>());
```

Comment:
I think `GET _alias/` will return JSON with the same amount of data, and it will be easier to work with - like with objects, not strings. Try to do that.

Reply:
done